### ISSUE TITLE
Allow to move subtrees

### ISSUE THREAD

**Issue by Drakulix** [iwburns/id-tree#1]
Removing and reattaching sub-trees to a different parent is currently not very easy, with only insert and remove operations available.

I would like to have something like `set_as_parent_and_child` as public interface, so I can change the parent of an existing node and move it without touching its children (like I have to when doing remove and re-insert).

Would you accept a pull request implementing this feature?
Anything I should be aware of when implementing this? I would basically wrap `set_as_parent_and_child` with `is_valid_node_id` and expose that.

**Comment by iwburns** [comment]
First of all, thanks so much for submitting an issue!  I think this is actually a feature that I wanted to add at one point, but I put it off (for one reason or another) and haven't tried to implement it since then.

Just so I'm 100% clear on what you're going for, would transitioning from a state like this:
![capture](https://cloud.githubusercontent.com/assets/6172641/21151043/ba05890a-c126-11e6-9a63-3accc9d8aef3.PNG)
to a state like this:
![capture1](https://cloud.githubusercontent.com/assets/6172641/21151054/c8cd0634-c126-11e6-84f2-cb7aa59b3f11.PNG)
be the behavior that you're after?
In other words, the sub-tree won't move to a different `Tree`, it will just be moved to a different parent within the original `Tree`.

**Comment by Drakulix** [comment]
Yes exactly.
Basically just changing the parent of an existing node given its id.
Sorry if I was sounding confusing, english is not my first language.

Maybe some context to clear things up:
I want to use your library to implement a tiling window manager, because I got tired of my own crappy implementation using Rc<RefCell<Node>> as children.
I really like the idea of the tree owning all the nodes and the id's still make it easy to reference them.
But for the window manager functionality I need to be able to swap and move some subtrees around. Basically all Leafs represent a Window and the upper nodes define how to split the remaining screen space to all children.

**Comment by iwburns** [comment]
Alright, very cool. That all makes sense to me.

Also, don't worry about your english.  That wasn't why I was clarifying.  I just wanted to be safe and make sure that I understood what you were going for.  Plus your english is much better than the english of many native speakers that I've heard!

To answer your original question: Yes, I would absolutely accept a pull request implementing this feature!

As for things to look out for when implementing this:

- be wary of the non-public functions inside `impl Tree`.  It may or may not be obvious how they're supposed to behave at first glance (I tried to comment where necessary, but if you have questions please feel free to ask and suggest improvements if you think of any)
- it would be good to consider how adding this feature is going to affect the advice given in the [Potential `NodeId` Issues](https://docs.rs/id_tree/0.2.4/id_tree/struct.NodeId.html#potential-nodeid-issues) section.  If we need to adjust that advice, or change the way we approach this feature to play nicely with the way `NodeId`s behave, we can do that as well.
- think about what to do if the `NodeId`s provided aren't valid.  Right now I'm thinking this function should return a `Result<(), NodeIdError>`, but I'm open to other suggestions as well.

Also, please include the following in your PR as well:

- documentation for the new function (in a format similar to the existing documentation)
- tests wherever appropriate to show that the new function works as expected.

If you have any questions at all feel free to ask!

**Comment by Drakulix** [comment]
Sure thing.

I will post any questions that might pop up here and open a pull request, once its done.
